lib/newrelic_httprb/instrumentation.rb

```diff
@@ -0,0 +1,33 @@
+DependencyDetection.defer do
+  named :http_rb
+
+  depends_on do
+    defined?(HTTP) && defined?(HTTP::Client)
+  end
+
+  executes do
+    ::NewRelic::Agent.logger.info 'Installing http.rb instrumentation'
+    require 'new_relic/agent/cross_app_tracing'
+    require 'newrelic_http/wrappers'
+  end
+
+  executes do
+    class HTTP::Client
+      def perform_with_newrelic_trace(request, options)
+        wrapped_request = ::NewRelicHTTP::HTTPRequest.new(request)
+
+        ::NewRelic::Agent::CrossAppTracing.tl_trace_http_request(wrapped_request) do
+          # RUBY-1244 Disable further tracing in request to avoid double
+          # counting if connection wasn't started (which calls request again).
```

Comment:
I'm planning on doing so and contribute a PR to their gem as well. However, running on a forked version for our systems makes me nervous. Maybe we can use it to see if the implementation passes on their test harness and use the gem anyway? Most of the contrib gems for newrelic that we use don't have *any* tests...